V1/utils/config.py

```diff
@@ -97,36 +276,52 @@ def init():
     global gui
     
     global log
-    
+    global path_to_output
+    global path_to_workload
+    global verbosity
     
     data = load_config()
     
+    # add verbosity and gui setting to config.py
+
     time = Time()
     event_queue = EventQueue()
 
+    # add try/catch for input validation
+
     task_types, task_type_names = create_task_types(data['task_types'])
     machine_types, machine_type_names, no_of_machines = create_machine_types(data['machines'])
     machines = []
 
-    capacity = data['battery'][0]['capacity']
-    total_energy = capacity * 3600
-    available_energy = total_energy
+    set_capacity(data['battery'][0]['capacity'])
 
-    machine_queue_size = data['parameters'][0]['machine_queue_size']
-    batch_queue_size = data['parameters'][0]['batch_queue_size']        
-    fairness_factor = data['parameters'][0]['fairness_factor']
-    bandwidth = data['cloud'][0]['bandwidth']
-    network_latency = data['cloud'][0]['network_latency']
+    set_machine_queue_size(data['parameters'][0]['machine_queue_size'])
+    # machine_queue_size = int(data['parameters'][0]['machine_queue_size'])
+    set_batch_queue_size(data['parameters'][0]['batch_queue_size'])     
+    set_fairness_factor(data['parameters'][0]['fairness_factor'])
+    set_bandwidth(data['cloud'][0]['bandwidth'])
+    set_network_latency(data['cloud'][0]['network_latency'])
 
     set_scheduler(data['parameters'][0]['scheduling_method'])
+    
 
     settings = data['settings'][0] 
-    gui = settings['gui'] 
+    set_gui(settings['gui']) 
+    set_verbosity(settings['verbosity'])
+    path_to_output = settings['path_to_output']
+    path_to_workload = settings['path_to_workload']
+
+    try:
+        log = open(f"{path_to_output}/log.txt",'w')
+    except OSError as err:
+        print(err)
+        path_to_output = "./output"
 
     try:
-        log = open(f"{settings['path_to_output']}/log.txt",'w')
+        log = open(f'{path_to_workload}/log.txt', 'w')
```

Comment:
Delete this block; we do not need log.txt in the workload directory; by the way, it overrides the previous log assignment.